src/CodeGen_ARM.cpp

```diff
@@ -705,68 +705,61 @@ void CodeGen_ARM::visit(const Store *op) {
             args[i] = codegen(call->args[i]);
         }
 
-        // Declare the function
-        std::ostringstream instr;
-        vector<llvm::Type *> arg_types;
-        if (target.bits == 32) {
-            instr << "llvm.arm.neon.vst"
-                  << num_vecs
-#if LLVM_VERSION > 37
-                   << ".p0i8"
-#endif
-                  << ".v"
-                  << intrin_type.lanes()
-                  << (t.is_float() ? 'f' : 'i')
-                  << t.bits();
-            arg_types = vector<llvm::Type *>(num_vecs + 2, llvm_type_of(intrin_type));
-            arg_types.front() = i8_t->getPointerTo();
-            arg_types.back() = i32_t;
-        } else {
-            instr << "llvm.aarch64.neon.st"
-                  << num_vecs
-                  << ".v"
-                  << intrin_type.lanes()
-                  << (t.is_float() ? 'f' : 'i')
-                  << t.bits()
-                  << ".p0"
-                  << (t.is_float() ? 'f' : 'i')
-                  << t.bits();
-            arg_types = vector<llvm::Type *>(num_vecs + 1, llvm_type_of(intrin_type));
-            arg_types.back() = llvm_type_of(intrin_type.element_of())->getPointerTo();
-        }
-        llvm::FunctionType *fn_type = FunctionType::get(llvm::Type::getVoidTy(*context), arg_types, false);
-        llvm::Function *fn = dyn_cast_or_null<llvm::Function>(module->getOrInsertFunction(instr.str(), fn_type));
-        internal_assert(fn);
-
-        // How many vst instructions do we need to generate?
-        int slices = t.lanes() / intrin_type.lanes();
+        Value *shufflePart1, *shufflePart2;
```

Comment:
Halide style would suggest these (and other local variables) be named shuffle_part1 and shuffle_part2.